gips/data/sentinel1/sentinel1.py

```diff
@@ -0,0 +1,417 @@
+        tilefile is called /archive/sentinel1/stage/tile_{}_{}.shp
+        """
+        outdir = "/archive/sentinel1/stage"
+        tileid_pattern = "{}-{}"
+        outname = "tiles.shp"
+        featurepath = os.path.join(outdir, 'feature.shp')
+        write_feature(vector, featurepath)
+        tilelist = make_tilegrid(featurepath, outdir, outname, tileid_pattern, cls._tile_attribute)
+
+        # TODO: use temp file
+        for file in glob.glob(os.path.splitext(featurepath)[0] + '.*'):
+            os.remove(file)
+
+        cls._tilefile_name = os.path.join(outdir, outname)
+        print('cls._tilefile_name', cls._tilefile_name)
+        return {k: (1,1) for k in tilelist}
+
+
+class sentinel1Asset(Asset):
+    Repository = sentinel1Repository
+    _sensors = {
+        'S1': {
+            'description': 'Sentinel-1, Satellite',
+            'band-strings':
+                ['01', '02'],
+            'polarization':
+                ('VV', 'VH'),
+            # band location in GHz
+            'bandlocs':
+                ['5.405', '5.405'],
+            'spatial-resolutions':
+                [20, 20],
+        },
+    }
+
+    _asset_pat_base = 'S1_IW_GRDH_(?P<year>\d{4})(?P<mon>\d{2})(?P<day>\d{2})_(?P<tile>\w{3}-\w{3})\.tif'
+
+    _start_date = datetime.date(2015, 1, 1)
+
+    _assets = {
+        'L1': {
+            'source': 'esa',
+            # 'pattern' is used for searching the repository of locally-managed assets
+            'pattern': _asset_pat_base,
+            'startdate': _start_date,
+            'latency': 3,
+        },
+    }
+
+    # default resultant resolution for resampling during to Data().copy()
+
+    def __init__(self, filename):
+        """Inspect a single file and set some metadata.
+        """
+        super(sentinel1Asset, self).__init__(filename)
+
+        self.basename = os.path.basename(filename)
+        match = re.match(self._asset_pat_base, self.basename)
+
+        self.asset = "L1"
+        self.sensor = "S1"
+
+        self.tile = match.group('tile')
+        self.date = datetime.date(*[int(i) for i in match.group('year', 'mon', 'day')])
+        self._version = int(''.join(
+            match.group('year', 'mon', 'day')
+        ))
+        self.meta = {} # for caching asset metadata values
+        self.tile_meta = None # tile metadata; dict was not good for this
+
+
+    @classmethod
+    def query_gs(cls, tile, date):
+        raise Exeption('query_gs not implemented')
+
+    @classmethod
+    def query_esa(cls, tile, date):
+        import gips.data.sentinel1.sentinel_api.sentinel_api as api
+
+        # use username and password for ESA DATA Hub authentication
+        username = cls.get_setting('username')
+        password = cls.get_setting('password')
+
+        # All historic Sentinel-1 scenes: https://scihub.copernicus.eu/dhus/
+        downloader = api.SentinelDownloader(username, password,
+            api_url='https://scihub.copernicus.eu/apihub/')
```

Comment:
What are the rate limits to downloading using their API?  Is it something ridiculous like 10 downloads at at time?

Reply:
Good question. All I know is that it seems slow.